### PR TITLE
https://issues.redhat.com/browse/ACM-7111 Train 06 agent reg API

### PR DESCRIPTION
**Pull request by oafischer**
For now, the new import via endpoint doc is a new file/topic, but I think it should go into the existing import via CLI file as a subtopic. Please advise.

### CODE REVIEW

File: clusters/cluster_lifecycle/import_agent.adoc [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+To import a managed cluster by using the the agent registration endpoint, complete the following steps:
+
+. Get the agent registration server URL by running the following command on the hub cluster:
++
+----
+export agent_registration_host=$(oc get route -n multicluster-engine agent-registration -o=jsonpath="{.spec.host}")
+----
++
+*Note:* If your hub is using a cluster-wide-proxy, make sure using the URL that managed cluster can access.
```
Comment (cadawson) [inline clusters/cluster_lifecycle/import_agent.adoc]:
```suggestion
*Note:* If your hub cluster is using a cluster-wide-proxy, make sure that you are using the URL that managed cluster can access.
```

File: clusters/cluster_lifecycle/import_agent.adoc [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+. Get the cacert by running the following command:
++
+----
+oc get configmap -n kube-system kube-root-ca.crt -o=jsonpath="{.data['ca\.crt']}" > ca.crt_
+----
+
+. Get the token for the agent regsitration sever to authorize by applying the following YAML content:
```
Comment (cadawson) [inline clusters/cluster_lifecycle/import_agent.adoc]:
```suggestion
. Get the token for the agent registration sever to authorize by applying the following YAML content:
```

File: clusters/cluster_lifecycle/import_agent.adoc [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    namespace: multicluster-engine
+----
+
+. Run the following command to export the token:
++
+----
+export token=$(oc get secret -n multicluster-engine managed-cluster-import-agent-registration-sa-token -o=jsonpath='{.data.token}' | base64 -d)
+----
+
+. Enable to automatic approval and patch the content to `cluster-manager` by running the following command:
```
Comment (cadawson) [inline clusters/cluster_lifecycle/import_agent.adoc]:
```suggestion
. Enable the automatic approval and patch the content to `cluster-manager` by running the following command:
```

File: clusters/cluster_lifecycle/import_agent.adoc [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
++
+----
+oc patch clustermanager cluster-manager --type=merge -p '{"spec":{"registrationConfiguration":{"featureGates":[
+{"feature": "ManagedClusterAutoApproval", "mode": "Enable"}], "autoApproveUsers":["system:serviceaccount:multicluster-engine:agent-registration-bootstrap"]}}}'
+----
++
+*Note:* You can also disable automatic approval and manually approve certificate signing requests from managed clusters.
+
+. Switch to your managed cluster and get the cacert by running the following command. Replace
```
Comment (cadawson) [inline clusters/cluster_lifecycle/import_agent.adoc]: Something is missing here, I think.

File: clusters/cluster_lifecycle/import_agent.adoc [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
++
+----
+curl --cacert ca.crt -H "Authorization: Bearer $token" https://$agent_registration_host/agent-registration/crds/v1 | oc apply -f -
+----
+
+. Run the following command to import the managed cluster to the hub cluster:
++
+Replace `<clusterName>` with the name of you cluster.
++
+*Optional*: Replace `<klusterletconfigName>` with the name of your KlusterletConfig.
```
Comment (cadawson) [inline clusters/cluster_lifecycle/import_agent.adoc]:
```suggestion
*Optional:* Replace `<klusterletconfigName>` with the name of your KlusterletConfig.
```

File: clusters/cluster_lifecycle/import_gui.adoc [hunk 6]
```diff
@@ -189,6 +189,33 @@ metadata:
...
+    tolerations:
+      - key: dedicated
+        operator: Equal
+        value: acm
+        effect: NoSchedule
+----
++
+<1> The name of your `KlusterletConfig`.
+
+. Add the `agent.open-cluster-management.io/klusterlet-config: `<klusterletconfigName>` annotation to the managed cluster, replacing `<klusterletconfigName>` with the name of your `KlusterletConfig`
```
Comment (cadawson) [inline clusters/cluster_lifecycle/import_gui.adoc]:
```suggestion
. Add the `agent.open-cluster-management.io/klusterlet-config: `<klusterletconfigName>` annotation to the managed cluster, replacing `<klusterletconfigName>` with the name of your `KlusterletConfig`.
```

File: clusters/cluster_lifecycle/specify_img_registry.adoc [hunk 7]
```diff
@@ -56,6 +54,27 @@ registries:
...
+    namespace: <pullSecretNamespace>
+    name: <pullSecretName>
+  registries:
+    - mirror: <mirrored-image-registry-address>
+      source: <image-registry-address>
+    - mirror: <mirrored-image-registry-address>
+      source: <image-registry-address>
+----
+ 
+See link:../cluster_lifecycle/import_agent.adoc#importing-managed-cluster-agent[Importing a managed cluster by using the agent registration endpoint] to learn more.
```
Comment (cadawson) [inline clusters/cluster_lifecycle/specify_img_registry.adoc]:
```suggestion
See xref:../cluster_lifecycle/import_agent.adoc#importing-managed-cluster-agent[Importing a managed cluster by using the agent registration endpoint] to learn more.
```

**Review by cadawson — CHANGES_REQUESTED** [review]
Looks pretty good. A couple of things to fix, but nothing major. Thanks!

**Comment by cadawson** [comment]
I think adding it as a subtopic of CLI could work.

**Comment by oafischer** [comment]
We discussed in a meeting a agreed to move forward with the third import option. Merging PR after final review.

**Review by cadawson — APPROVED** [review]
Looks good to me. Thanks, Oliver!

**Pull request merged**